src/main/flight/alt_hold.c

```diff
@@ -0,0 +1,274 @@
+
+    // things that always need to happen in the background should go here
+    altHoldState.measuredAltitudeCm = getAltitude();
+    DEBUG_SET(DEBUG_ALTHOLD, 1, (lrintf)(altHoldState.measuredAltitudeCm));
+
+    // exit unless altHold is active
+    if (!altHoldState.altHoldActive) {
+        DEBUG_SET(DEBUG_ALTHOLD, 0, 0);
+        return;
+    }
+
+
+    // ** for testing purposes... **
+
+    // estimate the vertical velocity from vertical accelerometer
+    // not used at present, but logged.
+    // my testing doesn't support using Accelerometer data for altitude control
+    // simple Z axis value might be OK when altHold is active mode since, unless windy, quad is close to level
+    t_fp_vector accelerationVector = {{
+        acc.accADC[X],
+        acc.accADC[Y],
+        acc.accADC[Z]
+    }};
+    imuTransformVectorBodyToEarth(&accelerationVector);
+    float measuredAccel = 9.8f * (accelerationVector.V.Z - acc.dev.acc_1G) / acc.dev.acc_1G; // m/s^2
+
+    // integrate into velocityFromAcc
+    // note that on commencing we may need to set the initial integral value to something derived from baro/GPS
+    altHoldState.velocityFromAcc += measuredAccel * taskIntervalSeconds; // m/s
+
+    // apply a leaky integrator or lowpass of time constant approx 10s (from original code *0.999)
+    // modified to leak with time constant approx 1s at 100Hz - can be detailed better later.
+    altHoldState.velocityFromAcc *= 0.99f;
+
+    // smooth the accelerometer values with its PT2
+    measuredAccel = pt2FilterApply(&altHoldAccVerticalLpf, measuredAccel);
+
+    // log at 100x for resolution
+    DEBUG_SET(DEBUG_ALTHOLD, 3, (lrintf)(measuredAccel * 100.0f)); // cm/s^2
+    DEBUG_SET(DEBUG_ALTHOLD, 4, (lrintf)(altHoldState.velocityFromAcc * 100.0f)); // cm/s * 10 ??
+
+    // ** end vertical acceleration stuff **
+
+
+    // check if the user has modified the target altitude using sticks while hovering
+    if (altholdConfig()->altHoldTargetAdjustRate) {
+        altHoldUpdateTargetAltitude();
+    }
+
+    // use PIDs to return the throttle adjustment value, and add it to the throttle value to use
+    const float throttleAdjustment = altitudePidCalculate();
+
+    float newThrottle = altholdConfig()->altHoldThrottleHover + throttleAdjustment;
+    altHoldState.throttleOut = constrainf(newThrottle, altholdConfig()->altHoldThrottleMin, altholdConfig()->altHoldThrottleMax);
+
+    DEBUG_SET(DEBUG_ALTHOLD, 0, (lrintf)(altHoldState.targetAltitudeCm));
+    DEBUG_SET(DEBUG_ALTHOLD, 2, (lrintf)(throttleAdjustment));
+}
+
+void initAltHoldState(void) {
+    altHoldInit();
+}
+
+void updateAltHoldState(timeUs_t currentTimeUs) {
+    altHoldUpdate();
+    UNUSED(currentTimeUs);
+}
+
+float altHoldGetThrottle(void) {
+    return scaleRangef(altHoldState.throttleOut, MAX(rxConfig()->mincheck, PWM_RANGE_MIN), PWM_RANGE_MAX, 0.0f, 1.0f);
+}
+
+bool altHoldIsActive(void){
+    return altHoldState.altHoldActive;
```

Comment:
Thanks.  I like to keep the 'altHold' in the name in case we want to find all instances where the altHoldActive is used... changed to isAltHoldActive.